kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/write/row/AdaptiveBigQueryWriter.java

```diff
@@ -74,24 +80,32 @@ public AdaptiveBigQueryWriter(BigQuery bigQuery,
   public void performWriteRequest(PartitionedTableId tableId,
                                   List<InsertAllRequest.RowToInsert> rows,
                                   String topic) {
-    InsertAllRequest request = createInsertAllRequest(tableId, rows);
-    InsertAllResponse writeResponse = bigQuery.insertAll(request);
-    // Should only perform one schema update attempt; may have to continue insert attempts due to
-    // BigQuery schema updates taking up to two minutes to take effect
-    if (writeResponse.hasErrors()
-        && onlyContainsInvalidSchemaErrors(writeResponse.getInsertErrors())) {
-      try {
-        schemaManager.updateSchema(tableId.getBaseTableId(), topic);
-      } catch (BigQueryException exception) {
-        throw new BigQueryConnectException("Failed to update table schema for: " + tableId.getBaseTableId(), exception);
+    InsertAllResponse writeResponse = null;
+    InsertAllRequest request = null;
+
+    try {
+      request = createInsertAllRequest(tableId, rows);
+      writeResponse = bigQuery.insertAll(request);
+      // Should only perform one schema update attempt; may have to continue insert attempts due to
+      // BigQuery schema updates taking up to two minutes to take effect
+      if (writeResponse.hasErrors()
+              && onlyContainsInvalidSchemaErrors(writeResponse.getInsertErrors())) {
+        attemptSchemaUpdate(tableId, topic);
+      }
+    } catch (BigQueryException e) {
+      if (isTableMissingSchema(e)) {
+        attemptSchemaUpdate(tableId, topic);
+      } else {
+        throw e;
       }
     }
 
     // Schema update might be delayed, so multiple insertion attempts may be necessary
     int attemptCount = 0;
-    while (writeResponse.hasErrors()) {
+    while (writeResponse == null || writeResponse.hasErrors()) {
       logger.trace("insertion failed");
-      if (onlyContainsInvalidSchemaErrors(writeResponse.getInsertErrors())) {
+      if (writeResponse == null || onlyContainsInvalidSchemaErrors(writeResponse.getInsertErrors())) {
+        // If the table was missing its schema, we never received a writeResponse
         logger.debug("re-attempting insertion");
         writeResponse = bigQuery.insertAll(request);
```

Comment:
we should have a try/catch around this, right? because we could get an exception of we have a null/empty schema? It's fine to just catch it and do nothing, but we should make sure the error doesn't percolate up if we intend to just retry.

Reply:
Good catch, added!